File: back-end/src/main/java/es/ull/project/application/usecase/operation/OperationUpdateUseCase.java
package es.ull.project.application.usecase.operation;

import java.util.Optional;

import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;
import es.ull.utils.docker.UllDockerImageName;

public interface OperationUpdateUseCase {
    /**
     * Update an operation
     * 
     * @param operation
     * @param operationName
     * @param dockerName
     * @return operation updated
     */
    Optional<Operation> updateOperation(OperationName oldOperationName, OperationName newOperationName, UllDockerImageName dockerImage);
}

File: front-end/src/application/usecase/operation/operation-update-use-case.ts
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';
import { Operation } from '../../../domain/value-object/operation';
import { OperationName } from '../../../domain/value-object/operation-name';
import { UllDockerImageName } from '@ull-tfg/ull-tfg-typescript/dist/ull-docker-image-name';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { NodeIcon } from '@/domain/enumerate/node-icon';

export interface OperationUpdateUseCase {
    /**
     * Update an operation
     * @param oldOperationName
     * @param operationName
     * @param dockerImage
     * @param icon
     * @returns operation updated
     */
    updateOperation(
        oldOperationName: OperationName,
        operationName: OperationName,
        dockerImage: UllDockerImageName,
    ): Promise<Either<DataError, Operation>>;
}
